src/custom/api/gnosisProtocol/hooks.ts

```diff
@@ -0,0 +1,15 @@
+import useSWR from 'swr'
+
+import { useActiveWeb3React } from 'hooks/web3'
+import { getOrders, OrderMetaData } from 'api/gnosisProtocol/api'
+import { AMOUNT_OF_ORDERS_TO_FETCH } from 'constants/index'
+
+export function useApiOrders(account?: string | null): OrderMetaData[] | undefined {
+  const { chainId } = useActiveWeb3React()
+
+  const { data } = useSWR<OrderMetaData[]>(['orders', account, chainId], () =>
```

Comment:
How much would this be caching? 

Should this have something like `refreshInterval`?

Reply:
We definitely can https://swr.vercel.app/docs/revalidation

I started without it to make it simpler and avoid potential race conditions.

How often do you think we should refresh it?

The main reason why we'd want to refresh is in case there is more than one instance of the page open in different browsers/devices.

In my view this is not a common use case to to need this.
Anyway, I'll set it to `30s` for testing and we can re-evaluate.

Reply:
https://github.com/gnosis/cowswap/pull/1951